corehq/apps/dropbox/views.py

```diff
@@ -0,0 +1,41 @@
+import os
+
+from dropbox.client import DropboxOAuth2Flow, DropboxClient
+
+from django.views.generic import View
+from django.http import HttpResponseRedirect, HttpResponse
+
+from .utils import get_dropbox_auth_flow
+
+DROPBOX_ACCESS_TOKEN = 'dropbox_access_token'
+
+
+class DropboxAuthInitiate(View):
+    slug = 'dropbox_auth_initiate'
+
+    def get(self, request, *args, **kwargs):
+        authorize_url = get_dropbox_auth_flow(request.session).start()
+        return HttpResponseRedirect(authorize_url)
+
+
+# TODO AuthCallback
+class DropboxAuthFinalize(View):
+    slug = 'dropbox_auth_finalize'
+
+    def get(self, request, *args, **kwargs):
+        try:
+            access_token, user_id, url_state = \
+                    get_dropbox_auth_flow(request.session).finish(request.GET)
+        except DropboxOAuth2Flow.BadRequestException, e:
+            HttpResponse(status=400)
+        except DropboxOAuth2Flow.BadStateException, e:
+            # Start the auth flow again.
+            return HttpResponseRedirect(reverse(DropboxAuthInitiate.slug))
+        except DropboxOAuth2Flow.CsrfException, e:
+            return HttpResponse(status=403)
+        except DropboxOAuth2Flow.NotApprovedException, e:
+            return HttpResponseRedirect("/", status=400)
+        except DropboxOAuth2Flow.ProviderException, e:
+            return HttpResponse(status=403)
+        request.session[DROPBOX_ACCESS_TOKEN] = access_token
```

Comment:
right now we store the token in the session. i did a bit of reading into this, and there are different strategies, `redis`, the user session, database, etc. one downside of using the database is that it's sensitive information that we'd be store in plain text. one downside of session is that, well, it's only stored in the session so we'd have to keep getting the access token if the session was wiped